### PR TITLE
Production Access

### PR DESCRIPTION
**Pull request by grrilla**
I want to do some research, and this is in much need of refinement. Points should be rephrased into a way that is demonstrable prior to being granted access. The opening is probably far too stream of consciousness and romantic to be usable.

### CODE REVIEW

File: competencies/production-access.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Competency - Production Access Readiness
+
+Control over access and permissions are in place for everyone's well-being, from the employee all the way back to the end-user. Having access to Production environments will make investigations, deployments, maintenance, and systems management faster, but also comes with a great deal of responsibility. It is important to understand the potential harm a single command can do while working in a production environment. A team member who is ready for Production access will know and consider the symptoms an action may have and take measures to isolate such actions wherever possible.
+
+To borrow an adage from Spiderman lore, "with great power comes great responsibility". For some reason, I'm also reminded of my favourite Yeats poem, "Never Give All the Heart", with the final couplet being: "He that made this knows all the cost, / For he gave all his heart and lost." Those who have made careless or ignorant mistakes in Production endure enough anxiety and cause enough damage that one would do better to follow a strict code than learn Production-safe conduct at the expense of themself and others.
```
Comment (codeallthethingz) [inline competencies/production-access.md:5]: A little self indulgent. :) 

```suggestion
"with great power comes great responsibility" - The Spiderman
```

File: competencies/production-access.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+## How do you prove it?
+
+You pair while working on Production systems. Always.
```
Comment (codeallthethingz) [inline competencies/production-access.md:9]: This is just something you can say.  How do you actually prove to a manager that you've paired and done this?   Be concrete in your prove it sections.

```suggestion
You have shadowed three production systems accesses.
```

File: competencies/production-access.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+You confirm and discuss actions with your pair partner such that both understand what you are attempting to do, why, and what the risks and alternatives are.
```
Comment (codeallthethingz) [inline competencies/production-access.md:11]: This should be in the how you do it section. See if you can think of a concrete, provable replacement.

File: competencies/production-access.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+You treat the super- and root user role with the utmost respect, only using them when absolutely necessary and avoiding them wherever possible.
```
Comment (codeallthethingz) [inline competencies/production-access.md:13]:
```suggestion
You can describe N common problems and attack vectors if using root.
```

File: competencies/production-access.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+## How do you prove it?
+
+You pair while working on Production systems. Always.
+
+You confirm and discuss actions with your pair partner such that both understand what you are attempting to do, why, and what the risks and alternatives are.
+
+You treat the super- and root user role with the utmost respect, only using them when absolutely necessary and avoiding them wherever possible.   
+
+You interact with databases using extreme levels of caution, understanding that corrupted and lost data can and will be absolutely breaking to our system. You are naturally averse to performing manual mutative actions to table data or DB state. You log in using a role with the fewest grants required to do your work. You do not execute complex queries on live databases, and exhaustively limit SELECT statements and other consumptive actions. You understand what a table scan is, why it is dangerous, and what actions will cause one to occur. You know which actions are locking at any level of the data (row, table, etc.) and understand the implications of performing those actions on a live database.
```
Comment (codeallthethingz) [inline competencies/production-access.md:15]: make concrete.

File: competencies/production-access.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+You treat Redis and any other datastore with the same degree of respect and caution as stated above.
+
+You download large log files and inspect them locally instead of on the host machine.
```
Comment (codeallthethingz) [inline competencies/production-access.md:19]: same

**Comment by olmobuining** [comment]
Maybe we can add other competencies as prerequisites. That might help with being able to prove this competency.
Suggestions for prerequisites:

- Linux (to maybe expand this into more levels or competencies, to include for example the large file transfer)
- SQL (maybe also expand this one to include level 2 of understanding table locks and caution required for certain queries)
- Security
- Git/Github
- Technical Breadth AWS Services

**Pull request merged**